### PR TITLE
[Bug] New Terms Rule Import Failing

### PR DESCRIPTION
**Pull request by eric-forte-elastic**
## Issues
https://github.com/elastic/detection-rules/issues/3542

## Summary

This PR is an intermediate fix for addressing an issue where new terms rules do not import successfully. See the above issue for more context.

There are two primary issues addressed in this PR:
1. The new terms rule schemas do not match what is supplied by Kibana
    * Our intermediate fix is to add a handler in the rule creation/importing loop that makes slight changes to the incoming data to support our schema.
2. The current schema for rule loading does not support having brackets in the rule name, where Kibana does.
    * Out fix for this is to direct add support for brackets in our schema to better support Kibana. 
    
    
## Testing 

To test these updates use the following two example new terms rules and run `python -m detection_rules import-rules <filename> --required-only` 

This first export tests the first issue independently of the second. 
[rules_export.ndjson.txt](https://github.com/elastic/detection-rules/files/14858586/rules_export.ndjson.txt)

This second export tests both combined. 
[rules_export_new_terms_test_other.ndjson.txt](https://github.com/elastic/detection-rules/files/14858585/rules_export_new_terms_test_other.ndjson.txt)

### CODE REVIEW

File: detection_rules/schemas/definitions.py [hunk 1]
```diff
@@ -17,7 +17,7 @@
 DATE_PATTERN = r'^\d{4}/\d{2}/\d{2}$'
 MATURITY_LEVELS = ['development', 'experimental', 'beta', 'production', 'deprecated']
 OS_OPTIONS = ['windows', 'linux', 'macos']
-NAME_PATTERN = r'^[a-zA-Z0-9].+?[a-zA-Z0-9()]$'
+NAME_PATTERN = r'^[a-zA-Z0-9].+?[a-zA-Z0-9\[\]()]$'
```
Comment (brokensound77) [inline detection_rules/schemas/definitions.py:20]: Have an example of why this was needed?

File: detection_rules/cli_utils.py [hunk 2]
```diff
@@ -165,7 +165,19 @@ def rule_prompt(path=None, rule_type=None, required_only=True, save=True, verbos
             contents[name] = schema_prompt(name, value=kwargs.pop(name))
             continue
 
-        result = schema_prompt(name, is_required=name in required_fields, **options.copy())
+        if name == "new_terms":
```
Comment (brokensound77) [inline detection_rules/cli_utils.py:168]: lgtm if it works 😓

**Review by brokensound77 — APPROVED** [review]
Ideally we need to fix the schema itself and modify the TOML state of the data (which I know we are punting for a bug fix a bit later).

This prompt is in desperate need of refactor, so intermediate patches help, but should be considered the temporary solution.

Nice job on this - LGTM 🚢

File: detection_rules/schemas/definitions.py (hunk 1 above)
Comment (eric-forte-elastic) [reply detection_rules/schemas/definitions.py:20]: Yes, any case where rules exported from Kibana have a title similar to `Rule Name [Duplicate]` 
Example from the testing block above that has this:

[rules_export_new_terms_test_other.ndjson.txt](https://github.com/elastic/detection-rules/files/14858585/rules_export_new_terms_test_other.ndjson.txt)

File: detection_rules/cli_utils.py [hunk 3]
```diff
@@ -165,7 +165,19 @@ def rule_prompt(path=None, rule_type=None, required_only=True, save=True, verbos
...
+            # patch to allow new_term imports
+            result = {"field": "new_terms_fields"}
+            if "new_terms_fields" in kwargs:
+                result["value"] = schema_prompt("new_terms_fields", value=kwargs.pop("new_terms_fields"))
+            if "history_window_start" in kwargs:
+                result["history_window_start"] = [
+                    {"field": "history_window_start",
+                     "value": schema_prompt("history_window_start", value=kwargs.pop("history_window_start"))}
+                ]
```
Comment (Mikaayenson) [inline detection_rules/cli_utils.py:177]: I see we're checking here for the fields history_window_start, but based on our schema if new terms is used, these fields are not optional. e.g. you can't have the field new_terms_field without history_window_start.

We may to still be able to prompt for the fields if they are not in kwargs.

Comment (eric-forte-elastic) [reply detection_rules/cli_utils.py]: Done

File: detection_rules/cli_utils.py [hunk 4]
```diff
@@ -165,7 +165,21 @@ def rule_prompt(path=None, rule_type=None, required_only=True, save=True, verbos
             contents[name] = schema_prompt(name, value=kwargs.pop(name))
             continue
 
-        result = schema_prompt(name, is_required=name in required_fields, **options.copy())
+        if name == "new_terms":
+            # patch to allow new_term imports
+            result = {"field": "new_terms_fields"}
+            if "new_terms_fields" in kwargs:
+                result["value"] = schema_prompt("new_terms_fields", value=kwargs.pop("new_terms_fields"))
```
Comment (Mikaayenson) [inline detection_rules/cli_utils.py:172]:
```suggestion
            result["value"] = schema_prompt("new_terms_fields", value=kwargs.pop("new_terms_fields"))
```

**Review by Mikaayenson — APPROVED** [review]
This LGTM. I do think we don't need this if check on `new_terms_fields`. The schema identifies that its `new_terms`, in which case all the fields are required.

**Pull request merged**